servers/zms/src/test/java/com/yahoo/athenz/zms/DBServiceTest.java

```diff
@@ -7012,4 +7012,72 @@ public void testAutoApplySolutionTemplate() {
 
         zms.deleteTopLevelDomain(mockDomRsrcCtx, domainName, auditRef);
     }
+
+    @Test
+    public void testApplySolutionTemplateWithRoleMetaData() {
+
+        String domainName = "solutiontemplate-rolemeta";
+        String caller = "testApplySolutionTemplateWithRoleMetaData";
+        TopLevelDomain dom1 = createTopLevelDomainObject(domainName,
+                "Test Domain1", "testOrg", adminUser);
+        zms.postTopLevelDomain(mockDomRsrcCtx, auditRef, dom1);
+
+        // apply the template
+
+        List<String> templates = new ArrayList<>();
+        templates.add("templateWithRoleMeta");
+        DomainTemplate domainTemplate = new DomainTemplate().setTemplateNames(templates);
+        zms.dbService.executePutDomainTemplate(mockDomRsrcCtx, domainName, domainTemplate, auditRef, caller);
+
+        DomainTemplateList domainTemplateList = zms.dbService.listDomainTemplates(domainName);
+        assertEquals(1, domainTemplateList.getTemplateNames().size());
+
+        // verify that our role collection includes the expected roles
+
+        List<String> names = zms.dbService.listRoles(domainName);
+        assertEquals(2, names.size());
+        assertTrue(names.contains("admin"));
+        assertTrue(names.contains("vip_admin"));
+
+        Role role = zms.dbService.getRole(domainName, "vip_admin", false, false, false);
+        assertEquals(domainName + ":role.vip_admin", role.getName());
+        assertNull(role.getTrust());
```

Comment:
in our test, we should verify that all attributes have been set that are defined in the template to make sure all attributes are correctly processed.

Reply:
Done